pkg/osvscanner/osvscanner.go

```diff
@@ -790,7 +824,7 @@ func DoScan(actions ScannerActions, r reporter.Reporter) (models.VulnerabilityRe
 			r.Errorf("Failed to resolved path with error %s\n", err)
 			return models.VulnerabilityResults{}, err
 		}
-		pkgs, err := scanLockfile(r, lockfilePath, parseAs)
+		pkgs, err := scanLockfile(r, lockfilePath, parseAs, !actions.CompareLocally)
```

Comment:
nit: it's a bit confusing that scanLockfile takes the inverse of actions.CompareLocally, while scanDir doesn't.

For consistency, it might be better for both to just accept actions.CompareLocally directly, and handle the logic of inverting that within scanLockfile?

Reply:
This also should be checking `actions.CompareOffline` as that's different from comparing locally, and I don't think there's a reason why the new resolver couldn't be used if you're comparing locally.

Reply:
transitive dependency resolution requires network access for two purposes:
 - to generate effective pom.xml: this may require fetching other pom.xml files from upstream
 - to call deps.dev resolver: the resolver sends requests to deps.dev API for requirements 

so I don't think you can use the resolver locally or offline.